zwave_js_server/client.py

```diff
@@ -83,6 +83,7 @@ async def connect(self) -> None:
         except (
             client_exceptions.WSServerHandshakeError,
             client_exceptions.ClientError,
+            TypeError,  # TypeError if the client receives non string
```

Comment:
I think the more correct way is to move fetching version info out of this try…catch block and make a helper `receive_json_or_raise` with [the same logic we use in `listen`](). We can then use that for version message, state message and for receiving events.

https://github.com/home-assistant-libs/zwave-js-server-python/blob/master/zwave_js_server/client.py#L152-L172